Client/Frontend/Browser/BrowserLongPressHandler.swift

```diff
@@ -0,0 +1,152 @@
+        return true
+    }
+
+    // MARK: - Long Press Gesture Handling
+    func didLongPress(gestureRecognizer: UILongPressGestureRecognizer) {
+        if gestureRecognizer.state == UIGestureRecognizerState.Began {
+            //Finding actual touch location in webView
+            var touchLocation = gestureRecognizer.locationInView(webView)
+            touchLocation.x -= webView!.scrollView.contentInset.left
+            touchLocation.y -= webView!.scrollView.contentInset.top
+            touchLocation.x /= webView!.scrollView.zoomScale
+            touchLocation.y /= webView!.scrollView.zoomScale
+
+            evaluateTouchLocation(touchLocation, gestureRecognizer: gestureRecognizer)
+        }
+    }
+
+    ///Check whether long pressed is on valid link. If found valid link then find the rect for link.
+    func evaluateTouchLocation(touchLocation: CGPoint,gestureRecognizer: UILongPressGestureRecognizer) {
+
+        self.webView!.evaluateJavaScript("findElementsAtPoint(\(touchLocation.x),\(touchLocation.y))", completionHandler: { (elementsJSON: AnyObject!, error: NSError!) -> Void in
+
+            if elementsJSON == nil || (elementsJSON as? String) == nil {
+                return
+            }
+
+            let elementsStr = elementsJSON as String
+            let json = JSON(string: elementsStr)
+            let tags = json["tags"].asArray
+
+            if tags == nil || tags?.count == 0 {
+                return
+            }
+
+            let selectedTags = tags!.filter({ ($0.asString == "A" || $0.asString == "IMG") })
+            if (selectedTags.count == 0) {
+                return
+            }
+            //
+            let hrefElement = json["hrefElement"].asDictionary
+            var elements:[ElementType] = []
+            if hrefElement != nil {
+                let hrefStr: String = hrefElement!["hrefLink"]!.asString!
+                let hrefLinkURL = NSURL(string: hrefStr)
+                let linkRect = self.rectLocationInWebView(self.webView!, locationRect:CGRectFromString(hrefElement!["location"]!.asString!))
+                if self.isValidURL(hrefLinkURL) {
+                    elements.append(ElementType.hrefLink(url: hrefLinkURL!, location: linkRect))
+                }
+            }
+
+            let imageElement = json["imageElement"].asDictionary
+            if imageElement != nil {
+                let imageSrcStr: String = imageElement!["imageSrc"]!.asString!
+                let imageSrc = NSURL(string: imageSrcStr)
+                let linkRect = self.rectLocationInWebView(self.webView!, locationRect:CGRectFromString(imageElement!["location"]!.asString!))
+                if self.isValidURL(imageSrc) {
+                    elements.append(ElementType.ImageSrc(url: imageSrc!, location: linkRect))
+                }
+            }
+
+            var disableGestures:[UIGestureRecognizer] = []
```

Comment:
@thebnich  I have tried all the things but it seems not possible to avoid default `longPressHandler` with old `document.body.style.webkitTouchCallout='none'` with `WKWebView`.Someone already posted on dev forums https://devforums.apple.com/message/1024729#1024729